src/test/java/wooteco/subway/path/PathTest.java

```diff
@@ -0,0 +1,117 @@
+package wooteco.subway.path;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static wooteco.subway.line.LineAcceptanceTest.지하철_노선_등록되어_있음;
+import static wooteco.subway.line.SectionAcceptanceTest.지하철_구간_등록되어_있음;
+import static wooteco.subway.station.StationAcceptanceTest.지하철역_등록되어_있음;
+
+import java.util.Arrays;
+import java.util.List;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import wooteco.subway.AcceptanceTest;
+import wooteco.subway.line.dto.LineResponse;
+import wooteco.subway.path.application.PathService;
+import wooteco.subway.path.dto.PathResponse;
+import wooteco.subway.station.dto.StationResponse;
+
+public class PathTest extends AcceptanceTest {
+
+    private LineResponse expectedShortestPath;
+    private LineResponse expectedNoShortestPath;
+    private StationResponse sourceStation1;
+    private StationResponse shortestStation2;
+    private StationResponse noShortestStation1;
+    private StationResponse shortestStation3;
+    private StationResponse shortestStation4;
+    private StationResponse destinationStation1;
+
+    @Autowired
+    private PathService pathService;
+
+    /**
+     * 교대역    --- *2호선* ---   강남역 | | *3호선*                   *신분당선* | | 남부터미널역  --- *3호선* ---   양재
+     */
+    @BeforeEach
+    public void setUp() {
```

Comment:
Q. 패러미터 주입은 해당 객체를 Bean에서 찾아서 주입시켜 주는 것이고 생성자 주입은 해당 패러미터를 인자로 넣어서 해당 인자를 넣어 주는 게 아닌가요? 결과적으로는 같은 동작을 한다고 판단했는데 아닌가요?
A. 결론부터 말씀드리자면 프로덕션 코드에서의 구현과 테스트 코드에서의 구현이 다르기 때문에 발생한 차이라고 볼 수 있습니다
테스트코드의 경우 JUnit에서 @BeforeEach에 맞게 바인딩 해주는데, 메서드 레벨에 @Autowired를 작성 시 바인딩해주지 않는 것이 문제입니다!
@BeforeEach에서 넘겨받기 위해선 

```suggestion
    public void setUp(@Autowired final PathService pathService) {
```

매개변수에 @Autowired를 걸어주시면 됩니다 😄 

Q. JPA가 왜 빈 생성자를 포함하는가? 에 대한 학습은 아직 하지 않았어요. DB 관련 에러가 계속 발생하기는 하지만 아직 JPA 학습을 하지 않고 사용하지도 않았기 때문에 문제가 없다고 생각했는데 계속 이 부분에 대해서 참고해야 할 것 처럼 스택 오버플로우에 질문과 응답이 많은 것 같아요. 즉,
Spring 동작 과정에서 빈 생성자의 의미 를 중점으로 찾으면 문제를 해결 할 수 있을까요?
A. 해당 문제는 JPA에서 어떠한 방식으로 객체를 생성하는가? 라는 의문으로 시작하여 리플렉션에 대한 이해가 있어야 빈 생성자가 있어야 하는 이유에 대해 알 수 있을 것 같습니다 :)
문제에 따라 접근하는 방법이 많기 때문에..! 검색하는 스킬을 높이는 것이 가장 좋다는 생각이 드네요 T.T

Reply:
좋은 답변 감사합니다!
세부적인 구현이 어떻게 되는지에 대해서 이해가 더 필요할 것 같아요.
조금 더 공부해보겠습니다!